modules/assignment/src/main/java/com/intuit/wasabi/assignment/cache/impl/AssignmentsMetadataCacheImpl.java

```diff
@@ -0,0 +1,356 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.intuit.wasabi.assignment.cache.impl;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.intuit.wasabi.assignment.cache.AssignmentsMetadataCache;
+import com.intuit.wasabi.experimentobjects.Application;
+import com.intuit.wasabi.experimentobjects.BucketList;
+import com.intuit.wasabi.experimentobjects.Experiment;
+import com.intuit.wasabi.experimentobjects.Page;
+import com.intuit.wasabi.experimentobjects.PageExperiment;
+import com.intuit.wasabi.experimentobjects.PrioritizedExperimentList;
+import com.intuit.wasabi.repository.CassandraRepository;
+import com.intuit.wasabi.repository.ExperimentRepository;
+import com.intuit.wasabi.repository.MutexRepository;
+import com.intuit.wasabi.repository.PagesRepository;
+import com.intuit.wasabi.repository.PrioritiesRepository;
+import net.sf.ehcache.Cache;
+import net.sf.ehcache.CacheManager;
+import net.sf.ehcache.Element;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.isNull;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ *  Local cache created and used during user assignment flow.
+ *
+ */
+
+public class AssignmentsMetadataCacheImpl implements AssignmentsMetadataCache {
+    private final Logger LOGGER = LoggerFactory.getLogger(AssignmentsMetadataCacheImpl.class);
+
+    private ExperimentRepository experimentRepository = null;
+    private PrioritiesRepository prioritiesRepository = null;
+    private MutexRepository mutexRepository = null;
+    private PagesRepository pagesRepository = null;
+    private AssignmentsMetadataCacheRefreshTask metadataCacheRefreshTask;
+    private CacheManager cacheManager;
+
+    private static final String ASSIGNMENT_METADATA_CACHE_SERVICE_NAME = "AssignmentsMetadataCache";
+
+    @Inject
+    public AssignmentsMetadataCacheImpl(@CassandraRepository ExperimentRepository experimentRepository, PrioritiesRepository prioritiesRepository,
+                                        MutexRepository mutexRepository, PagesRepository pagesRepository,
+                                        @Named("AssignmentsMetadataCacheRefreshCacheService") ScheduledExecutorService refreshCacheService,
```

Comment:
I am with you on this... But it would be odd to create constants for some of the inject names rather than for all the inject names. May be, we can work on this in separate PR to have a place to define all the Guice names/constants and use them across - that will avoid any typos and code consistency.